sdk/identity/Azure.Identity/src/PersistentTokenCache.cs

```diff
@@ -16,44 +16,45 @@ public class PersistentTokenCache : TokenCache
-        private static AsyncLockWithValue<MsalCacheHelper> s_ProtectedCacheHelperLock = new AsyncLockWithValue<MsalCacheHelper>();
-        private static AsyncLockWithValue<MsalCacheHelper> s_FallbackCacheHelperLock = new AsyncLockWithValue<MsalCacheHelper>();
-        private readonly bool _allowUnencryptedStorage;
-        private readonly string _name;
-
-        /// <summary>
-        /// Creates a new instance of <see cref="PersistentTokenCache"/>.
-        /// </summary>
-        /// <param name="allowUnencryptedStorage"></param>
-        public PersistentTokenCache(bool allowUnencryptedStorage = true)
-        {
-            _allowUnencryptedStorage = allowUnencryptedStorage;
-        }
+        internal static readonly string s_msalCacheClientId = Guid.NewGuid().ToString();
+        internal readonly bool _allowUnencryptedStorage;
+        internal readonly string _name;
+        private static AsyncLockWithValue<MsalCacheHelperWrapper> cacheHelperLock = new AsyncLockWithValue<MsalCacheHelperWrapper>();
+        private readonly MsalCacheHelperWrapper _cacheHelperWrapper;
 
         /// <summary>
         /// Creates a new instance of <see cref="PersistentTokenCache"/> with the specified options.
         /// </summary>
         /// <param name="options">Options controlling the storage of the <see cref="PersistentTokenCache"/>.</param>
-        public PersistentTokenCache(PersistentTokenCacheOptions options)
+        public PersistentTokenCache(PersistentTokenCacheOptions options = null)
         {
             _allowUnencryptedStorage = options?.AllowUnencryptedStorage ?? false;
+            _name = options?.Name ?? Constants.DefaultMsalTokenCacheName;
+            _cacheHelperWrapper = new MsalCacheHelperWrapper(false);
+        }
 
-            _name = options?.Name;
+        internal PersistentTokenCache(PersistentTokenCacheOptions options, MsalCacheHelperWrapper cacheHelperWrapper)
+        {
+            _allowUnencryptedStorage = options?.AllowUnencryptedStorage ?? false;
+            _name = options?.Name ?? Constants.DefaultMsalTokenCacheName;
+            _cacheHelperWrapper = cacheHelperWrapper;
         }
 
         internal override async Task RegisterCache(bool async, ITokenCache tokenCache, CancellationToken cancellationToken)
         {
-            MsalCacheHelper cacheHelper = await GetCacheHelperAsync(async, cancellationToken).ConfigureAwait(false);
+            MsalCacheHelperWrapper cacheHelper = await GetCacheHelperAsync(async, cancellationToken).ConfigureAwait(false);
 
             cacheHelper.RegisterCache(tokenCache);
 
             await base.RegisterCache(async, tokenCache, cancellationToken).ConfigureAwait(false);
         }
 
-        private async Task<MsalCacheHelper> GetCacheHelperAsync(bool async, CancellationToken cancellationToken)
+        internal static void ResetWrapperCache()
+        {
+            cacheHelperLock = new AsyncLockWithValue<MsalCacheHelperWrapper>();
```

Comment:
does this also need to remove the helper from the MSAL cache to stop it from handling events?

Reply:
It's purely for testing purposes so that tests can exercise the code paths more than once. It should never be used in conjunction with the real MSAL cache. Unless I'm missing some other scenario where this is important, perhaps just a comment would be helpful to address this concern?

Reply:
Discussed more offline - added a comment.